src/Workspaces/Core/Portable/FindSymbols/FindReferences/Finders/AbstractMemberScopedReferenceFinder.cs

```diff
@@ -68,26 +68,25 @@ protected sealed override bool CanFind(TSymbol symbol)
         {
             for (var current = symbol; current != null; current = current.ContainingSymbol)
             {
+                if (current.DeclaringSyntaxReferences.Length == 0)
+                    continue;
+
                 if (current is IPropertySymbol)
                     return current;
 
                 // If this is an initializer for a property's backing field, then we want to 
                 // search for results within the property itself.
                 if (current is IFieldSymbol field)
                 {
-                    if (field.IsImplicitlyDeclared &&
-                        field.AssociatedSymbol?.Kind == SymbolKind.Property)
-                    {
-                        return field.AssociatedSymbol;
-                    }
-                    else
-                    {
-                        return field;
-                    }
+                    return field is { IsImplicitlyDeclared: true, AssociatedSymbol.Kind: SymbolKind.Property }
+                        ? field.AssociatedSymbol
+                        : field;
                 }
 
-                if (current is IMethodSymbol { MethodKind: not MethodKind.AnonymousFunction and not MethodKind.LocalFunction } method)
-                    return method;
+                // Note: this may hit a containing local-function/lambda.  That's fine as that's still the scope we want
+                // to look for this local within.
+                if (current is IMethodSymbol)
+                    return current;
```

Comment:
we were skipping over containing methods before, finding the topmost method to search in.  This just means we have more to look in.  A local cannot be references outside the method it was defined in (whether that method is an 'ordinary' method, a 'local function' or a 'lambda'), so this just caused us to do more work.

Generally, this would only matter in very large methods.  But it's still worthwhile to do to trim things down.  Found when doing the work in https://github.com/dotnet/roslyn/pull/67324